src/benchmark_environments/util.py

```diff
@@ -17,3 +22,183 @@ def step(self, action):
             info["terminal_observation"] = obs
             obs = self.env.reset()
         return obs, rew, False, info
+
+
+class AbsorbAfterDoneWrapper(gym.Wrapper):
+    """Goes to absorb state after done=True. Always returns done=False.
+
+    A convenient way to add absorbing states to environments like MountainCar.
+    """
+
+    def __init__(
+        self,
+        env: gym.Env,
+        absorb_reward: float = 0.0,
+        absorb_obs: Optional[np.ndarray] = None,
+    ):
+        """Initialize AbsorbAfterDoneWrapper.
+
+        Args:
+          env: The wrapped Env.
+          absorb_reward: The reward returned at the absorb state.
+          absorb_obs: The observation returned at the absorb state. If None, then
+            repeat the final observation before absorb.
+        """
+        super().__init__(env)
+        self.absorb_reward = absorb_reward
+        self.absorb_obs_default = absorb_obs
+        self.absorb_obs_this_episode = None
+        self.at_absorb_state = None
+
+    def reset(self, *args, **kwargs):
+        """Reset the environment."""
+        self.at_absorb_state = False
+        self.absorb_obs_this_episode = None
+        return self.env.reset(*args, **kwargs)
+
+    def step(self, action):
+        """Advance the environment by one step.
+
+        This wrapped `step()` always returns done=False.
+
+        After the first done is returned by the underlying Env, we enter an artificial
+        absorb state. In this artificial absorb state, we stop calling
+        `self.env.step(action)` (i.e. the `action` argument is entirely ignored) and
+        we return fixed values for obs, rew, done, and info.
+        """
+        if not self.at_absorb_state:
+            inner_obs, inner_rew, done, inner_info = self.env.step(action)
+            if done:
+                # Initialize the artificial absorb state, which we will repeatedly use
+                # starting on the next call to `step()`.
+                self.at_absorb_state = True
+
+                if self.absorb_obs_default is None:
```

Comment:
Inclined to make this a one-liner, although maybe it's too long...

```suggestion
                self.absorb_obs_this_episode = inner_obs if self.absorb_obs_default is None else self.absorb_obs_default
```

Reply:
It's too long (length 120). Tried breaking it up with "\", but I mildly prefer the current formatting:

![image](https://user-images.githubusercontent.com/1750835/76057234-bac71e00-5f2d-11ea-8a6e-530be6ac4bca.png)

Reply:
black doesn't accept breaking up with `\`. Automatically formats to
![image](https://user-images.githubusercontent.com/1750835/76122276-89406800-5faa-11ea-965a-75ce3882ba28.png)

I'm sticking to the original format.